src/c++/perf_analyzer/concurrency_worker.h

```diff
@@ -73,17 +74,16 @@ class ConcurrencyWorker : public LoadWorker {
       const bool using_json_data, const bool streaming,
       const SharedMemoryType shared_memory_type, const int32_t batch_size,
       std::vector<std::shared_ptr<ThreadConfig>>& threads_config,
-      std::vector<std::shared_ptr<SequenceStat>>& sequence_stat,
       std::unordered_map<std::string, SharedMemoryData>& shared_memory_regions,
       std::condition_variable& wake_signal, std::mutex& wake_mutex,
-      size_t& active_threads, bool& execute, std::atomic<uint64_t>& curr_seq_id,
-      std::uniform_int_distribution<uint64_t>& distribution)
+      size_t& active_threads, bool& execute,
+      std::shared_ptr<SequenceManager> sequence_manager)
       : LoadWorker(
-            id, thread_stat, parser, data_loader, factory, sequence_stat,
+            id, thread_stat, parser, data_loader, factory,
             shared_memory_regions, backend_kind, shared_memory_type,
             on_sequence_model, async, streaming, batch_size, using_json_data,
-            sequence_length, start_sequence_id, sequence_id_range, curr_seq_id,
-            distribution, wake_signal, wake_mutex, execute),
+            sequence_length, start_sequence_id, sequence_id_range, wake_signal,
```

Comment:
can sequence_length, start_sequence_id, and sequence_id_range be removed from most classes now? Do the workers consume this information?

Reply:
Removed 👍

Good lord... 18 files changed, 136 additions, 195 deletions, just to remove these three variables from being disseminated as they were...............................................................